target/linux/realtek/files-5.10/drivers/net/dsa/rtl83xx/rtl838x.c

```diff
@@ -512,6 +512,46 @@ static void rtl838x_l2_learning_setup(void)
 	sw_w32(0, RTL838X_SPCL_TRAP_ARP_CTRL);
 }
 
+static void rtl838x_enable_learning(int port, bool enable)
+{
+	// Limit learning to maximum: 32k entries, after that just flood (bits 0-1)
+
+	if (enable)  {
+		// flood after 32k entries
+		sw_w32((0x3fff << 2) | 0, RTL838X_L2_PORT_LRN_CONSTRT + (port << 2));
+	} else {
+		// just forward
+		sw_w32(0, RTL838X_L2_PORT_LRN_CONSTRT + (port << 2));
+	}
+}
+
+static void rtl838x_enable_flood(int port, bool enable)
+{
+	u32 flood_mask = sw_r32(RTL838X_L2_PORT_LRN_CONSTRT + (port << 2));
+
+	if (enable)  {
+		// flood
+		flood_mask &= ~3;
+		flood_mask |= 0;
+		sw_w32(flood_mask, RTL838X_L2_PORT_LRN_CONSTRT + (port << 2));
+	} else {
+		// drop (bit 1)
+		flood_mask &= ~3;
+		flood_mask |= 1;
+		sw_w32(flood_mask, RTL838X_L2_PORT_LRN_CONSTRT + (port << 2));
+	}
+}
+
+static void rtl838x_enable_mcast_flood(int port, bool enable)
+{
+
+}
```

Comment:
AFAICT you check that the function pointers these calls are stored in are not NULL, so you don't need these empty calls.

Reply:
I will leave it in. It is a good reminder that code is missing. Just leaving an empty pointer is not as obvious. Especially if you are wondering why something might not work...

Reply:
An empty function is hardly any more clear than a NULL pointer to indicate support is possible but not implemented. Please add a to do comment in that case. If you already have an idea of where to start, that would also be useful info to add.